connector/src/main/scala/com/microsoft/kusto/spark/utils/KustoClientCache.scala

```diff
@@ -0,0 +1,65 @@
+package com.microsoft.kusto.spark.utils
+
+import java.util.concurrent.ConcurrentHashMap
+
+import com.microsoft.azure.kusto.data.{ClientFactory, ConnectionStringBuilder}
+import com.microsoft.azure.kusto.ingest.IngestClientFactory
+import com.microsoft.kusto.spark.authentication.{AadApplicationAuthentication, KeyVaultAuthentication, KustoAccessTokenAuthentication, KustoAuthentication}
+import com.microsoft.kusto.spark.utils.{KustoConstants => KCONST}
+
+object KustoClientCache {
```

Comment:
I wouldn't call this a cache. it it more of a factory or a pool than a cache (caching is just an implementation detail). 
`KustoBackendFactory` or `KustoDatabasePool`

Reply:
Once we change KustoClient, we need to update this as well.